File: data/recomposed_projects/llama-3-70b-instruct/body/commons-graph/evosuite-test/src/test/org/apache/commons/graph/scc/DefaultSccAlgorithmSelector_ESTest.py
from __future__ import annotations
import time
import re
import os
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.graph.DirectedGraph import *
from src.main.org.apache.commons.graph.model.DirectedMutableGraph import *
from src.main.org.apache.commons.graph.model.RevertedGraph import *
from src.main.org.apache.commons.graph.scc.DefaultSccAlgorithmSelector import *

# from src.test.org.apache.commons.graph.scc.DefaultSccAlgorithmSelector_ESTest_scaffolding import *
# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *


class DefaultSccAlgorithmSelector_ESTest(unittest.TestCase):

    def test12(self) -> None:

        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            int, LinkedHashSet[int]
        ](None)

        try:
            defaultSccAlgorithmSelector0.applyingTarjan()
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.scc.TarjanAlgorithm", e)

    def test11(self) -> None:

        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            LinkedHashSet[int], int
        ](None)

        with self.assertRaises(RuntimeError):
            defaultSccAlgorithmSelector0.applyingKosarajuSharir0()
            self.fail("Expecting exception: RuntimeError")

        verifyException("org.apache.commons.graph.scc.KosarajuSharirAlgorithm", e)

    def test10(self) -> None:

        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            LinkedHashSet[int], LinkedHashSet[int]
        ](None)
        linkedHashSet0 = LinkedHashSet[int]()

        try:
            defaultSccAlgorithmSelector0.applyingKosarajuSharir1(linkedHashSet0)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.scc.KosarajuSharirAlgorithm", e)

    def test09(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[int, int](
            directedMutableGraph0
        )
        set0 = defaultSccAlgorithmSelector0.applyingCheriyanMehlhornGabow()
        self.assertTrue(len(set0) == 0)

    def test08(self) -> None:

        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            LinkedHashSet[int], int
        ](None)

        with pytest.raises(RuntimeError):
            defaultSccAlgorithmSelector0.applyingCheriyanMehlhornGabow()
            self.fail("Expecting exception: RuntimeError")

        verifyException(
            "org.apache.commons.graph.scc.CheriyanMehlhornGabowAlgorithm", e
        )

    def test07(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector(
            directedMutableGraph0
        )
        linkedHashSet0 = set()

        with self.assertRaises(RuntimeError):
            defaultSccAlgorithmSelector0.applyingKosarajuSharir1(linkedHashSet0)

        verifyException("org.apache.commons.graph.utils.Assertions", RuntimeError)

    def test06(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector(
            directedMutableGraph0
        )

        with pytest.raises(RuntimeError):
            defaultSccAlgorithmSelector0.applyingKosarajuSharir1(None)
            self.fail("Expecting exception: RuntimeError")

    def test05(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -1
        directedMutableGraph0.addVertex(integer0)
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[int, int](
            directedMutableGraph0
        )
        set0 = defaultSccAlgorithmSelector0.applyingCheriyanMehlhornGabow()
        self.assertFalse(set0)

    def test04(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[
            LinkedHashSet[int], LinkedHashSet[int]
        ]()
        linkedHashSet0 = LinkedHashSet[int]()
        directedMutableGraph0.addVertex(linkedHashSet0)
        revertedGraph0 = RevertedGraph[LinkedHashSet[int], LinkedHashSet[int]](
            directedMutableGraph0
        )
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            LinkedHashSet[int], LinkedHashSet[int]
        ](revertedGraph0)
        set0 = defaultSccAlgorithmSelector0.applyingKosarajuSharir0()
        self.assertEqual(1, len(set0))

    def test03(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, LinkedHashSet[int]]()
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            int, LinkedHashSet[int]
        ](directedMutableGraph0)
        set0 = defaultSccAlgorithmSelector0.applyingKosarajuSharir0()
        self.assertEqual(0, len(set0))

    def test02(self) -> None:

        # Create a DirectedMutableGraph
        directedMutableGraph0 = DirectedMutableGraph()

        # Create a LinkedHashSet
        linkedHashSet0 = LinkedHashSet()

        # Add the LinkedHashSet to the DirectedMutableGraph
        directedMutableGraph0.addVertex(linkedHashSet0)

        # Create a RevertedGraph
        revertedGraph0 = RevertedGraph(directedMutableGraph0)

        # Create a DefaultSccAlgorithmSelector
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector(revertedGraph0)

        # Apply the Kosaraju-Sharir algorithm
        set0 = defaultSccAlgorithmSelector0.applyingKosarajuSharir1(linkedHashSet0)

        # Assert that the set is not empty
        assert not set0.isEmpty()

    def test01(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, int]()
        integer0 = -1473
        directedMutableGraph0.addVertex(integer0)
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[int, int](
            directedMutableGraph0
        )
        set0 = defaultSccAlgorithmSelector0.applyingTarjan()
        self.assertFalse(set0.isEmpty())

    def test00(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph[int, LinkedHashSet[int]]()
        defaultSccAlgorithmSelector0 = DefaultSccAlgorithmSelector[
            int, LinkedHashSet[int]
        ](directedMutableGraph0)
        set0 = defaultSccAlgorithmSelector0.applyingTarjan()
        self.assertTrue(len(set0) == 0)

File: java_projects/cleaned_final_projects_evosuite/commons-graph/evosuite-tests/org/apache/commons/graph/scc/DefaultSccAlgorithmSelector_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.graph.scc;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.LinkedHashSet;
import java.util.Set;
import org.apache.commons.graph.DirectedGraph;
import org.apache.commons.graph.model.DirectedMutableGraph;
import org.apache.commons.graph.model.RevertedGraph;
import org.apache.commons.graph.scc.DefaultSccAlgorithmSelector;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class DefaultSccAlgorithmSelector_ESTest extends DefaultSccAlgorithmSelector_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      DirectedMutableGraph<Integer, LinkedHashSet<Integer>> directedMutableGraph0 = new DirectedMutableGraph<Integer, LinkedHashSet<Integer>>();
      DefaultSccAlgorithmSelector<Integer, LinkedHashSet<Integer>> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<Integer, LinkedHashSet<Integer>>(directedMutableGraph0);
      Set<Set<Integer>> set0 = defaultSccAlgorithmSelector0.applyingTarjan();
      assertTrue(set0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-1473));
      directedMutableGraph0.addVertex(integer0);
      DefaultSccAlgorithmSelector<Integer, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<Integer, Integer>(directedMutableGraph0);
      Set<Set<Integer>> set0 = defaultSccAlgorithmSelector0.applyingTarjan();
      assertFalse(set0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>> directedMutableGraph0 = new DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      directedMutableGraph0.addVertex(linkedHashSet0);
      RevertedGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>> revertedGraph0 = new RevertedGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>(directedMutableGraph0);
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, LinkedHashSet<Integer>> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, LinkedHashSet<Integer>>(revertedGraph0);
      Set<LinkedHashSet<Integer>> set0 = defaultSccAlgorithmSelector0.applyingKosarajuSharir1(linkedHashSet0);
      assertFalse(set0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      DirectedMutableGraph<Integer, LinkedHashSet<Integer>> directedMutableGraph0 = new DirectedMutableGraph<Integer, LinkedHashSet<Integer>>();
      DefaultSccAlgorithmSelector<Integer, LinkedHashSet<Integer>> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<Integer, LinkedHashSet<Integer>>(directedMutableGraph0);
      Set<Set<Integer>> set0 = defaultSccAlgorithmSelector0.applyingKosarajuSharir0();
      assertEquals(0, set0.size());
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>> directedMutableGraph0 = new DirectedMutableGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      directedMutableGraph0.addVertex(linkedHashSet0);
      RevertedGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>> revertedGraph0 = new RevertedGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>(directedMutableGraph0);
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, LinkedHashSet<Integer>> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, LinkedHashSet<Integer>>(revertedGraph0);
      Set<Set<LinkedHashSet<Integer>>> set0 = defaultSccAlgorithmSelector0.applyingKosarajuSharir0();
      assertEquals(1, set0.size());
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      Integer integer0 = new Integer((-1));
      directedMutableGraph0.addVertex(integer0);
      DefaultSccAlgorithmSelector<Integer, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<Integer, Integer>(directedMutableGraph0);
      Set<Set<Integer>> set0 = defaultSccAlgorithmSelector0.applyingCheriyanMehlhornGabow();
      assertFalse(set0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      DirectedMutableGraph<LinkedHashSet<Integer>, Integer> directedMutableGraph0 = new DirectedMutableGraph<LinkedHashSet<Integer>, Integer>();
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer>(directedMutableGraph0);
      // Undeclared exception!
      try { 
        defaultSccAlgorithmSelector0.applyingKosarajuSharir1((LinkedHashSet<Integer>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // Kosaraju Sharir algorithm cannot be calculated without expressing the source vertex
         //
         verifyException("org.apache.commons.graph.utils.Assertions", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      DirectedMutableGraph<LinkedHashSet<Integer>, Integer> directedMutableGraph0 = new DirectedMutableGraph<LinkedHashSet<Integer>, Integer>();
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer>(directedMutableGraph0);
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      // Undeclared exception!
      try { 
        defaultSccAlgorithmSelector0.applyingKosarajuSharir1(linkedHashSet0);
        fail("Expecting exception: IllegalStateException");
      
      } catch(IllegalStateException e) {
         //
         // Vertex [] does not exist in the Graph
         //
         verifyException("org.apache.commons.graph.utils.Assertions", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer>((DirectedGraph<LinkedHashSet<Integer>, Integer>) null);
      // Undeclared exception!
      try { 
        defaultSccAlgorithmSelector0.applyingCheriyanMehlhornGabow();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.scc.CheriyanMehlhornGabowAlgorithm", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      DirectedMutableGraph<Integer, Integer> directedMutableGraph0 = new DirectedMutableGraph<Integer, Integer>();
      DefaultSccAlgorithmSelector<Integer, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<Integer, Integer>(directedMutableGraph0);
      Set<Set<Integer>> set0 = defaultSccAlgorithmSelector0.applyingCheriyanMehlhornGabow();
      assertTrue(set0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, LinkedHashSet<Integer>> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, LinkedHashSet<Integer>>((DirectedGraph<LinkedHashSet<Integer>, LinkedHashSet<Integer>>) null);
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      // Undeclared exception!
      try { 
        defaultSccAlgorithmSelector0.applyingKosarajuSharir1(linkedHashSet0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.scc.KosarajuSharirAlgorithm", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<LinkedHashSet<Integer>, Integer>((DirectedGraph<LinkedHashSet<Integer>, Integer>) null);
      // Undeclared exception!
      try { 
        defaultSccAlgorithmSelector0.applyingKosarajuSharir0();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.scc.KosarajuSharirAlgorithm", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      DefaultSccAlgorithmSelector<Integer, LinkedHashSet<Integer>> defaultSccAlgorithmSelector0 = new DefaultSccAlgorithmSelector<Integer, LinkedHashSet<Integer>>((DirectedGraph<Integer, LinkedHashSet<Integer>>) null);
      // Undeclared exception!
      try { 
        defaultSccAlgorithmSelector0.applyingTarjan();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.scc.TarjanAlgorithm", e);
      }
  }
}
